droidlet/dashboard/web/src/components/Timeline/TimelineUtils.js

```diff
@@ -49,3 +30,21 @@ export function jsonToArray(eventObj) {
   }
   return tableArr;
 }
+
+export function handleClick(stateManager, item) {
+  const eventObj = JSON.parse(item);
+  let tableArr = jsonToArray(eventObj);
+  stateManager.memory.timelineDetails = tableArr;
+  stateManager.updateTimeline();
+
+  var config = {
+    title: capitalizeEvent(eventObj["name"]),
+    cssClass: "scrollable",
+    type: "react-component",
+    component: "TimelineDetails",
+    props: { stateManager: stateManager },
+  };
+  stateManager.dashboardLayout.root.contentItems[0].contentItems[1].contentItems[0].contentItems[5].contentItems[1].contentItems[1].addChild(
```

Comment:
This is confusing. Please document it to clarify

Reply:
Just looked into it and found [a cleaner way of doing it](https://github.com/facebookresearch/droidlet/pull/600/files#diff-56fc3f8596cce6ec5f485c3e588443d7a05a6d879c00667ca2c32b33adaf8422R47), lmk how that looks!